internal/network/machine_network_cidr.go

```diff
@@ -516,25 +516,58 @@ func GetDefaultRouteNetworkByFamily(h *models.Host, networks map[AddressFamily][
 	return ret, fmt.Errorf("can not find cidr by route: no inventory for host %s", h.ID.String())
 }
 
-func IsHostInPrimaryMachineNetCidr(log logrus.FieldLogger, cluster *common.Cluster, host *models.Host) bool {
-	// The host should belong to all the networks specified as Machine Networks.
+// Parse the Machine Network CIDRs into IPNet
+func parseMachineNetworks(machineNetworks []*models.MachineNetwork) ([]*net.IPNet, error) {
+	var parsedCidr []*net.IPNet
+	for _, machineNet := range machineNetworks {
+		_, machineIpnet, err := net.ParseCIDR(string(machineNet.Cidr))
+		if err != nil {
+			return nil, err
+		}
+		parsedCidr = append(parsedCidr, machineIpnet)
+	}
 
+	return parsedCidr, nil
+}
+
+// Check if a host belongs to all the networks specified as Machine Networks.
+func IsHostInPrimaryMachineNetCidr(log logrus.FieldLogger, cluster *common.Cluster, host *models.Host) bool {
 	// TODO(mko) This rule should be revised as soon as OCP supports multiple machineNetwork
```

Comment:
let's check if it's ClusterManagedNetworking retrun true
cause we allow only one network

Reply:
ignore it - It OK
you can have different host network form ISCSI network